### PR TITLE
Add ovirt_datacenters and ovirt_datacenters_facts modules

### PR DESCRIPTION
**Pull request by machacekondra**
##### ISSUE TYPE

- New Module Pull Request
##### COMPONENT NAME

- ovirt_datacenters
- ovirt_datacenters_facts
##### ANSIBLE VERSION

```
ansible 2.1.1.0
```
##### SUMMARY

Add ovirt_datacenters and ovirt_datacenters_facts modules to manage oVirt datacenters.

```

```

### CODE REVIEW

**Comment by machacekondra** [comment]
@mwperina @ryansb

**Comment by machacekondra** [comment]
ready_for_review

**Comment by gregdek** [comment]
Thanks @machacekondra for this new module. When this module receives 'shipit' comments from two community members and any 'needs_revision' comments have been resolved, we will mark for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by machacekondra** [comment]
shipit

**Comment by gregdek** [comment]
Thanks @machacekondra. Since you are one of the maintainers (@machacekondra) of this module and you commented with "shipit", we are marking this PR for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

File: cloud/ovirt/ovirt_datacenters.py [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+
+EXAMPLES = '''
+# Examples don't contain auth parameter for simplicity,
+# look at ovirt_auth module to see how to reuse authentication:
+
+# Create datacenter
+- ovirt_datacenters:
+    name: mydatacenter
+    local: True
+    compatibility_version: 4.1
```
Comment (mwperina) [inline cloud/ovirt/ovirt_datacenters.py]: Please use version 4.0, as 4.1 may not be released before Ansible 2.3

**Review by mwperina — CHANGES_REQUESTED** [review]

File: cloud/ovirt/ovirt_datacenters_facts.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        - "Search term which is accepted by oVirt search backend."
+        - "For example to search datacenter I(X) use following pattern: I(name=X)"
+extends_documentation_fragment: ovirt
+'''
+
+EXAMPLES = '''
+# Examples don't contain auth parameter for simplicity,
+# look at ovirt_auth module to see how to reuse authentication:
+
+# Gather facts about all datacenters named C<production*>:
```
Comment (mwperina) [inline cloud/ovirt/ovirt_datacenters_facts.py]: I'd rephrase to: 'Gather facts about all data centers which names start with C<production>:'

**Review by mwperina — CHANGES_REQUESTED** [review]
Generally looks good, just 2 small doc issues

File: cloud/ovirt/ovirt_datacenters_facts.py [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+        - "Search term which is accepted by oVirt search backend."
+        - "For example to search datacenter I(X) use following pattern: I(name=X)"
+extends_documentation_fragment: ovirt
+'''
+
+EXAMPLES = '''
+# Examples don't contain auth parameter for simplicity,
+# look at ovirt_auth module to see how to reuse authentication:
+
+# Gather facts about all data centers which names start with C<production*>:
```
Comment (mwperina) [inline cloud/ovirt/ovirt_datacenters_facts.py]: 'C(production*)' -> 'C(production)'

**Review by mwperina — CHANGES_REQUESTED** [review]

File: cloud/ovirt/ovirt_datacenters_facts.py [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+        - "Search term which is accepted by oVirt search backend."
+        - "For example to search datacenter I(X) use following pattern: I(name=X)"
+extends_documentation_fragment: ovirt
+'''
+
+EXAMPLES = '''
+# Examples don't contain auth parameter for simplicity,
+# look at ovirt_auth module to see how to reuse authentication:
+
+# Gather facts about all data centers which names start with C<production>:
```
Comment (mwperina) [inline cloud/ovirt/ovirt_datacenters_facts.py]: `C<production>` -> `C(production)`

**Review by mwperina — CHANGES_REQUESTED** [review]

File: cloud/ovirt/ovirt_datacenters_facts.py (hunk 4 above)
Comment (machacekondra) [reply cloud/ovirt/ovirt_datacenters_facts.py]: done

**Review by mwperina — APPROVED** [review]
Looks good to me, thanks!

**Comment by gregdek** [comment]
Thanks again to @machacekondra. This module is going into community review.

We notice that you are one of the maintainers (@machacekondra) of this module, so when you think it's ready, please comment "shipit" and we will consider it for merging.

[This message brought to you by your friendly Ansibull-bot.]

File: cloud/ovirt/ovirt_datacenters.py [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+try:
+    import ovirtsdk4 as sdk
+    import ovirtsdk4.types as otypes
```
Comment (ryansb) [inline cloud/ovirt/ovirt_datacenters.py:25]: Please add a conditional on these (similar to the AWS modules) so your module doesn't raise ImportError at load.

**Comment by ryansb** [comment]
Looks good overall, just add a failure mode for when the oVirt SDK isn't installed.

**Comment by machacekondra** [comment]
shipit

**Comment by gregdek** [comment]
Thanks @machacekondra. Since you are one of the maintainers (@machacekondra) of this module and you commented with "shipit", we are marking this PR for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

**Review by mwperina — APPROVED** [review]
:+1:
